### ISSUE TITLE
KeyError: 'expanded_url' with minet twitter scrape tweets

### ISSUE THREAD

**Issue by lakonis** [medialab/minet#978]
I encounter the following error on a simple and common request : `minet twitter scrape tweets "since:2014-01-01 until:2014-01-31 from:myuser" --cookie firefox`

```bash
Scraping ━━━━━━━━━━━ 0/1 queries - in 27.80s (?/s)                                                                                                                                                                 
⠧ since:2014-01-01 until:2014-01-31 from:myuser 80/? tweets in 27.72s (2.89/s) total: 80 tweets                                                                                                                  
minet process was stopped because an error occurred!
Traceback (most recent call last):
  File "/home/user/githn/tweets/myuser/venv/lib/python3.11/site-packages/minet/twitter/api_scraper.py", line 636, in request_tweet_search
    result = normalize_tweet(
             ^^^^^^^^^^^^^^^^
  File "/home/user/githn/tweets/myuser/venv/lib/python3.11/site-packages/twitwi/normalizers.py", line 334, in normalize_tweet
    normalized = custom_normalize_url(entity['expanded_url'])
                                      ~~~~~~^^^^^^^^^^^^^^^^
KeyError: 'expanded_url'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/user/githn/tweets/myuser/venv/bin/minet", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/user/githn/tweets/myuser/venv/lib/python3.11/site-packages/minet/cli/__main__.py", line 14, in main
    run("minet", __identifier__, MINET_COMMANDS)
  File "/home/user/githn/tweets/myuser/venv/lib/python3.11/site-packages/minet/cli/utils.py", line 49, in wrapper
    fn(*args, **kwargs)
  File "/home/user/githn/tweets/myuser/venv/lib/python3.11/site-packages/minet/cli/run.py", line 139, in run
    fn(cli_args)
  File "/home/user/githn/tweets/myuser/venv/lib/python3.11/site-packages/minet/cli/utils.py", line 468, in wrapper
    return action(cli_args, *args, **additional_kwargs, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/githn/tweets/myuser/venv/lib/python3.11/site-packages/minet/cli/twitter/scrape.py", line 81, in action
    for data in iterator:
  File "/home/user/githn/tweets/myuser/venv/lib/python3.11/site-packages/minet/twitter/api_scraper.py", line 739, in search_tweets
    new_cursor, tweets = self.request_tweet_search(
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/githn/tweets/myuser/venv/lib/python3.11/site-packages/minet/web.py", line 1336, in decorated
    return retryer(fn, self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/githn/tweets/myuser/venv/lib/python3.11/site-packages/minet/web.py", line 1225, in __call__
    return super().__call__(fn, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/githn/tweets/myuser/venv/lib/python3.11/site-packages/tenacity/__init__.py", line 467, in __call__
    do = self.iter(retry_state=retry_state)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/githn/tweets/myuser/venv/lib/python3.11/site-packages/tenacity/__init__.py", line 368, in iter
    result = action(retry_state)
             ^^^^^^^^^^^^^^^^^^^
  File "/home/user/githn/tweets/myuser/venv/lib/python3.11/site-packages/tenacity/__init__.py", line 390, in <lambda>
    self._add_action_func(lambda rs: rs.outcome.result())
                                     ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/concurrent/futures/_base.py", line 449, in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/concurrent/futures/_base.py", line 401, in __get_result
    raise self._exception
  File "/home/user/githn/tweets/myuser/venv/lib/python3.11/site-packages/tenacity/__init__.py", line 470, in __call__
    result = fn(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^
  File "/home/user/githn/tweets/myuser/venv/lib/python3.11/site-packages/minet/twitter/api_scraper.py", line 643, in request_tweet_search
    raise TwitterPublicAPIParsingError(tweet["id_str"])
minet.twitter.exceptions.TwitterPublicAPIParsingError: 426318495450943488

```
Does it look like something usual ? I first thought it was an network or API problem, but it is recurrent each time I scrape this particular user at this specific time period. As if one tweet would crash the scrapping. If you are interested to investigate, I can provide the twitter username.

**Comment by Yomguithereal** [comment]
@lakonis I will need the twitter username to reproduce indeed.

**Comment by lakonis** [comment]
here it is: `@lmerzeau`

Let me know how it goes.

**Comment by Yomguithereal** [comment]
I am investigating, but first fun fact: the website itself crashes on the related tweet: https://x.com/lmerzeau/status/426318495450943488 (it displays "something went wrong", on my end at least)

**Comment by Yomguithereal** [comment]
So, long story short, some `t.co` links from Twitter are now broken and when this happens, the website's API returns misformatted payloads (lol), and the website itself breaks, which is quite funny I find. I have updated the scraper to handle this gracefully though. It should be fixed from v2.0.7 onward.

**Comment by lakonis** [comment]
Bravo, merci for the fix!
